pom.xml

```diff
@@ -196,6 +199,24 @@
         </exclusion>
       </exclusions>
     </dependency>
+    <dependency>
+      <!-- TODO maybe these dependency needs to be changed/removed -->
+      <groupId>io.atlassian.util.concurrent</groupId>
+      <artifactId>atlassian-util-concurrent</artifactId>
+      <version>4.0.1</version>
+    </dependency>
+    <dependency>
+      <!-- TODO maybe these dependency needs to be changed/removed -->
```

Comment:
we use a plugin to maintain same level of httpclient see
```
    <dependency>
      <groupId>org.jenkins-ci.plugins</groupId>
      <artifactId>apache-httpcomponents-client-4-api</artifactId>
      <version>4.5.5-3.0</version>
    </dependency>
```
Maybe check if we need upgrade or if we grab correct version with that?